src/app/FAKE/CommandlineParams.fs

```diff
@@ -4,18 +4,35 @@ open Fake
 
 let printAllParams() = printfn "FAKE.exe [buildScript] [Target] Variable1=Value1 Variable2=Value2 ... "
 
+(* 
+    This is a set of flags that exist in code lower in the target processing that MUST be normalized to a certain form. This is necessary because
+    If any old styole variables are present in the FAKE invocation, Argu parsing will fail and we will not have parsed out those commands correctly
+    from the overall command line.
+
+    You can typically find usages of 'hasBuildParam' in this codebase to find places where these values are required.
+*)
+let specialFlags = 
+    [
+        "-st", "single-target"
+        "--single-target", "single-target" 
+        "-pd", "details"
+        "--print-details", "details"
+    ] |> Map.ofList
+
 let parseArgs cmdArgs = 
-    let splitter = [| '=' |]
-    let split (arg:string) = 
-      let pos = arg.IndexOfAny splitter 
-      [| arg.Substring(0, pos); arg.Substring(pos + 1, arg.Length - pos - 1) |]
+    let (|KeyValue|Flag|TargetName|) ((i,arg) : int * string) =
```

Comment:
This is the same logic from the previous version, just captured into an AP to make it very clear what case is what.